scvi/model/base/_vaemixin.py

```diff
@@ -210,3 +213,45 @@ def get_latent_representation(
             if return_dist
             else torch.cat(latent).numpy()
         )
+
+    @torch.inference_mode()
+    def get_batch_representation(
+        self,
+        adata: AnnData | None = None,
+        batch_keys: list[str] | None = None,
+    ) -> np.ndarray:
+        """Return batch representations.
+
+        Parameters
+        ----------
+        adata
+            AnnData object with equivalent structure to initial AnnData. If ``None``, defaults to
+            the AnnData object used to initialize the model.
+        batch_keys
+            Keys for batches to return representations for. If ``None``, all batches are used.
+
+        Returns
+        -------
+        Array of batch representations of shape ``(n_batches, batch_embedding_dim)``.
+        """
+        self._check_if_trained(warn=False)
+
+        if not hasattr(self.module, "batch_embedding"):
+            raise NotImplementedError("Model does not support batch embeddings.")
+        if self.module.batch_embedding is None:
+            raise ValueError("Model was not trained with batch embeddings.")
+
+        adata = self._validate_anndata(adata)
+        manager = self.get_anndata_manager(adata)
+        batch_state_registry = manager.get_state_registry(REGISTRY_KEYS.BATCH_KEY)
+        cat_mapping = batch_state_registry.categorical_mapping
+
+        if batch_keys is None:
+            batch_indexes = np.arange(len(cat_mapping))
+        elif not all((key in cat_mapping) for key in batch_keys):
+            raise ValueError("``batch_keys`` contains keys not present in ``adata``.")
+        else:
+            batch_indexes = np.where(cat_mapping == batch_keys)[0]
+
+        batch_embeddings = self.module.batch_embedding.weight.detach().cpu().numpy()
```

Comment:
Potentially leave this on GPU. Run embedding and retrieve the embeddings. This object can be rather large if you have a large number of batches.

Reply:
Yeah it could potentially be large depending on the number of batches. Two reasons why I support moving it to CPU: (1) it would be more consistent with `get_latent_representation`, which returns numpy arrays, (2) and by default the batch embeddings are guaranteed to be smaller than the latent representation since `batch_embedding_dim=5`.

Reply:
We are not copying a 5-dimensional batch_embedding but a n_batchxbatch_embedding_dim dimensional tensor here. However, copying batch_indices to GPU will not reduce the amount that needs to be transferred.